floris/wind_data.py

```diff
@@ -452,25 +452,51 @@ def upsample(self, wd_step=None, ws_step=None, method="linear", inplace=False):
                 "Available methods are 'linear' and 'nearest'"
             )
 
+        # First establish the current ws_step and wd_step
+        if len(self.wind_speeds) >= 2:
+            ws_step_current = self.wind_speeds[1] - self.wind_speeds[0]
```

Comment:
Do we need something that checks that the `wind_speeds` and `wind_directions` are actually evenly spaced? It seems possible that a user could specify them with irregular intervals (probably an edge case, but could happen)? Perhaps we could just add a check to the initialization of the roses to ensure even spacing?

Reply:
Added above and to TI rose, only doubt I have is if we need such strictness for TI, but probably it will help avoid bugs where we use the first elements for step size

Reply:
The requirement of being evenly spaced should probably be added to the docstring argument descriptions.

Reply:
Adding now